starlette/routing.py

```diff
@@ -84,7 +84,7 @@ async def app(scope: Scope, receive: Receive, send: Send) -> None:
 
 
 def get_name(endpoint: typing.Callable) -> str:
-    if inspect.isfunction(endpoint) or inspect.isclass(endpoint):
+    if inspect.isroutine(endpoint) or inspect.isclass(endpoint):
```

Comment:
Does the behavior of lambda changes now?

Reply:
No:

```python3
import inspect
my_lambda = lambda: 1
assert inspect.isfunction(my_lambda) == inspect.isroutine(my_lambda)
```